client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -475,9 +503,61 @@ function getQuickstartUrl( {
 	if ( orderId ) {
 		return;
 	}
+
+	// This is used when we need to quickly dial back the volume of concierge sessions
+	// being offered and sold, to be inline with HE availability.
+	// Change this percentage to change the amount of offers given out:
+	const percentConciergeOffers = 75;
+
+	if ( ! badNaiveClientSideRollout( 'conciergeUpsellDial', percentConciergeOffers ) ) {
+		return;
+	}
+
 	return `/checkout/offer-quickstart-session/${ pendingOrReceiptId }/${ siteSlug }`;
 }
 
+function getProfessionalEmailUpsellUrl( {
+	pendingOrReceiptId,
+	cart,
+	siteSlug,
+	orderId,
+	domains,
+}: {
+	pendingOrReceiptId: string;
+	cart: ResponseCart | undefined;
+	siteSlug: string | undefined;
+	orderId: number | undefined;
+	domains: Domain[] | undefined;
+} ): string | undefined {
+	if ( orderId || ! cart ) {
+		return;
+	}
+
+	if (
+		! config.isEnabled( 'upsell/professional-email' ) ||
+		hasGoogleApps( cart ) ||
+		hasTitanMail( cart )
+	) {
+		return;
+	}
+
+	if ( hasPremiumPlan( cart ) ) {
+		return;
+	}
+
+	if ( ! hasBloggerPlan( cart ) && ! hasPersonalPlan( cart ) && ! hasBusinessPlan( cart ) ) {
```

Comment:
I think we should show this upsell to eCommerce plan purchasers as well.
```suggestion
	if ( ! hasBloggerPlan( cart ) && ! hasPersonalPlan( cart ) && ! hasBusinessPlan( cart ) && ! hasEcommercePlan( cart ) ) {
```

Reply:
Hmmm... We're not showing this to ECommerce plan purchasers. We likely need to update the PR description to match, as well as adding an explicit unit test to include/exclude this case.

Reply:
Right, I actually noticed that on Thursday but forgot to push that change 🤦‍♂️.

Reply:
I've updated the PR description to remove this -- we can add that in a quick follow-up PR.